src/frames.ts

```diff
@@ -428,8 +428,11 @@ export class Frame {
     let resolve: (error: Error|void) => void;
     const promise = new Promise<Error|void>(x => resolve = x);
     const watch = (documentId: string, error?: Error) => {
-      if (documentId !== expectedDocumentId)
-        return resolve(new Error('Navigation interrupted by another one'));
+      if (documentId !== expectedDocumentId) {
+        if (!error)
```

Comment:
if (documentId !== expectedDocumentId && !error)
  error = ...

Would be slightly easier to understand I believe.

Reply:
Reworked a bit, thanks.